Kuroba/app/src/main/java/com/github/adamantcheese/chan/core/manager/UpdateManager.java

```diff
@@ -256,41 +267,70 @@ private boolean processUpdateApiResponse(UpdateApiRequest.UpdateApiResponse resp
     public void doUpdate(UpdateApiRequest.UpdateApiResponse response) {
         BackgroundUtils.ensureMainThread();
 
-        fileCache.downloadFile(response.apkURL.toString(), new FileCacheListener() {
+        if (cancelableDownload != null) {
+            cancelableDownload.cancel();
+            cancelableDownload = null;
+        }
+
+        cancelableDownload = fileCacheV2.enqueueNormalDownloadFileRequest(response.apkURL.toString(), new FileCacheListener() {
             @Override
-            public void onProgress(long downloaded, long total) {
+            public void onProgress(int chunkIndex, long downloaded, long total) {
                 BackgroundUtils.ensureMainThread();
 
                 updateDownloadDialog.setProgress((int) (updateDownloadDialog.getMax() * (downloaded / (double) total)));
             }
 
             @Override
-            public void onSuccess(RawFile file) {
+            public void onSuccess(AbstractFile file) {
                 BackgroundUtils.ensureMainThread();
 
                 updateDownloadDialog.dismiss();
                 updateDownloadDialog = null;
+
+                // We need to remove all the dots from the file name because FSAF won't allow that.
+                String fileName = StringUtils.dirNameRemoveBadCharacters(
```

Comment:
But isn't any other app does the same? Like NewPipe does for example (https://github.com/TeamNewPipe/NewPipe/blob/38b0b79644abc9ef9efdee92fe5380e0f8117e0e/app/src/main/java/org/schabi/newpipe/util/FilenameUtils.java#L22). I just don't get what's the point of having them. And this will surely break FSAF because I check for %2E too.

Reply:
If the user selects CHARSET_MOST_SPECIAL in NewPipe, then periods are allowable in filenames. Either way, only RawFiles should ever be passed to this method. But I'm sure periods are allowable. By the way you don't check for %2E anywhere in your FSAF library, nor in Kuroba. I would recommend you add some tests in FSAF for those cases where periods are included in filenames.

Reply:
>internal const val ENCODED_SEPARATOR = "%2F"

Yeah, I've mixed it up, I'm checking for different kinds of the slash symbol, not periods.

Reply:
Not done yet. Gonna do this one tomorrow since it will take some time.

Reply:
Alright, periods in file names should work fine now. But not in directory names! I don't think we should allow periods in directory names.